src/pages/MapEditorPage/hooks/useCanvasInteractions.js

```
@@ -0,0 +1,449 @@
+            return
+        }
+
+        setActiveEditorTab(type)
+        if (type !== 'zone' || selectedEntity?.id !== id) {
+            setSelectedZoneVertexIndex(null)
+        }
+        setSelectedEntity({ type, id })
+    }
+
+    const viewportCursor = toolMode === 'pan'
+        ? (dragStateRef.current ? 'grabbing' : 'grab')
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

  253 |
  254 |     const viewportCursor = toolMode === 'pan'
> 255 |         ? (dragStateRef.current ? 'grabbing' : 'grab')
      |            ^^^^^^^^^^^^^^^^^^^^ Cannot access ref value during render
  256 |         : ((pendingGatePick || isAddingPoi || isAddingZone || isAddingNode || isAddingEdge || activeConnectorForMapping) ? 'crosshair' : 'default')
  257 |
  258 |     function createNodeAtPoint(point) {

(react-hooks/refs)